src/lib.rs

```diff
@@ -8,6 +8,8 @@ pub mod protocol;
 pub mod provider;
 pub mod rpc_protocol;
 
+pub mod net;
```

Comment:
Just for now, probably want to make this private once it is actually being used.